packages/@coorpacademy-components/scripts/observables/generate-stories.js

```diff
@@ -1,43 +1,81 @@
-const {join, relative} = require('path');
-const {map} = require('rxjs/operators');
-const {concat, of} = require('rxjs');
+const {join, dirname, relative} = require('path');
+const {of, from, concat} = require('rxjs');
+const {map, toArray, mergeMap} = require('rxjs/operators');
 const {readComponentFixtures$} = require('./component-fixtures');
 const {pascalCase} = require('./string');
 const {readComponents$} = require('./components');
+const {readAllStoryFiles$, deleteFolder$, removeEmptyFolders$} = require('./cleanup');
 
-const generateStories$ = cwd =>
-  readComponents$(cwd).pipe(
+/**
+ * Generate story files for all current components.
+ */
+const generateStories$ = cwd => {
+  // 1) Build [storiesPath, lines$] pairs for each component
+  const generation$ = readComponents$(cwd).pipe(
     map(({title, path, type, titleRaw, levels}) => {
       const testPath = join(path, 'test');
       const storiesPath = join(testPath, 'index.stories.tsx');
-      return [
-        storiesPath,
-        concat(
-          of(`import React from 'react';`, `import ${title} from '..';`),
-          readComponentFixtures$({title, path, type}).pipe(
-            map(
-              ({fixture, fixturePath}) =>
-                `import fixture${fixture} from './${relative(testPath, fixturePath)}';`
-            )
-          ),
-          of(
-            `
-export default {
-  title: '${[...levels.map(pascalCase), titleRaw].join('/')}',
-  component: ${title}
-};`
-          ),
-          readComponentFixtures$({title, path, type}).pipe(
-            map(
-              ({fixture}) =>
-                `
+
+      // Observable that emits lines for fixture imports
+      const fixtureImports$ = readComponentFixtures$({title, path, type}).pipe(
```

Comment:
```suggestion
const fixtures$ = readComponentFixtures$({title, path, type}).pipe(shareReplay(1));
```
ca t'évitera de read le filesystème pour avoir les fixtures deux fois